gm4/plugins/versioning.py

```diff
@@ -56,6 +57,26 @@ def modules(ctx: Context, opts: VersioningConfig):
         lines.append(f"execute if score {dep_id} load.status matches 1.. unless score {dep_id} load.status matches {dep_ver.major} run data modify storage gm4:log queue append value {log_data}")
         lines.append(f"execute if score {dep_id} load.status matches {dep_ver.major} unless score {dep_id}_minor load.status matches {dep_ver.minor}.. run data modify storage gm4:log queue append value {log_data}")
 
+    # add required environment checks
+    for namespaced_environment_check in opts.environment_checks:
```

Comment:
I almost think we should only run environment checks on first install, instead of every reload? Also, we only need to perform each check once. Putting them into load makes each module run the same check again. Maybe we can finally utilize pre_load for this kind of setup check?

Reply:
That would miss the edge case of importing an existing world to a new server.

Reply:
> Putting them into load makes each module run the same check again
That's why these checks leave a score behind which is only reset post load. If the check was already done this reload, it won't need to check again and just uses the last result. That result is invalidated in post load, making sure the check is run again once next reload.